Compiling kotlin files on Windows within Git Bash fails

These errors appear on the console:

```
[jbang] [0:450] Compile: C:\Users\tako\.jbang\cache\kotlincs\2.0.21\kotlinc\bin\kotlinc.bat -classpath C:\Users\tako\.m2\repository\log4j\log4j\1.2.17\log4j-1.2.17.jar;C:\Users\tako\.m2
\repository\org\jetbrains\kotlin\kotlin-stdlib\2.0.21\kotlin-stdlib-2.0.21.jar;C:\Users\tako\.m2\repository\org\jetbrains\annotations\13.0\annotations-13.0.jar -d C:\Users\tako\.jbang\cache\jars\classpath_log.kt.fedeeb108b22ef4054ad2d333f70cd5fecc141e776accc0920ea3874281d8055\classes C:\Projects\jbang\jbang\itests\classpath_log.kt
error: source entry is not a Kotlin file: C:\Users\tako\.m2\repository\org\jetbrains\kotlin\kotlin-stdlib\2.0.21\kotlin-stdlib-2.0.21.jar
error: source entry is not a Kotlin file: C:\Users\tako\.m2\repository\org\jetbrains\annotations\13.0\annotations-13.0.jar
```

This is because normal Windows paths are used even though we're running inside a bash shell.

Providing the steps below to reproduce the error.

Main.java
```java
public class Main {
  public static void main(String[] args) {
    System.out.println("Hi!!");
  }
}
```

```
c:\> jbang run Main.java
Hi!!

c:\> bash
$ jbang run Main.java
Hi!!

$ jbang cache clear
[jbang] Clearing cache for urls
[jbang] Clearing cache for jars
[jbang] Clearing cache for kotlincs
[jbang] Clearing cache for groovycs
[jbang] Clearing cache for scripts
[jbang] Clearing cache for stdins
[jbang] Clearing cache for deps

$ jbang run Main.java
[jbang] Building jar for Main.java...
Hi!!
```

Main.kt
```kotlin
fun main(args: Array<String>) {
    println("Hi!")
}
```

```
c:\> jbang run Main.kt
Hi!!

c:\> bash
$ jbang run Main.kt
Hi!!

$ jbang cache clear
[jbang] Clearing cache for urls
[jbang] Clearing cache for jars
[jbang] Clearing cache for kotlincs
[jbang] Clearing cache for groovycs
[jbang] Clearing cache for scripts
[jbang] Clearing cache for stdins
[jbang] Clearing cache for deps

$ jbang run Main.kt
[jbang] Downloading Kotlin 2.0.21. Be patient, this can take several minutes...
[jbang] Installing Kotlin 2.0.21...
[jbang] Resolving dependencies...
[jbang]    org.jetbrains.kotlin:kotlin-stdlib:2.0.21
[jbang] Dependencies resolved
[jbang] Building jar for Main.kt...
error: source entry is not a Kotlin file: C:\Users\wfouche\.m2\repository\org\jetbrains\annotations\13.0\annotations-13.0.jar
[jbang] [ERROR] Error during compile
[jbang] Run with --verbose for more details. The --verbose must be placed before the jbang command. I.e. jbang --verbose run [...]
```

--- a/src/main/java/dev/jbang/util/CommandBuffer.java
+++ b/src/main/java/dev/jbang/util/CommandBuffer.java
@@ -161,7 +161,7 @@ public class CommandBuffer {
 		// argument, this is close to impossible to handle on Windows.
 		// (see
 		// https://stackoverflow.com/questions/30157414/batch-argument-with-quotes-and-spaces)
-		if (shell != Util.Shell.bash && cmdNeedQuotesChars.matcher(arg).find()) {
+		if (Util.isWindows() && cmdNeedQuotesChars.matcher(arg).find()) {
 			arg = "\"" + arg + "\"";
 		}
 		return arg;

--- a/src/test/java/dev/jbang/util/TestCommandBuffer.java
+++ b/src/test/java/dev/jbang/util/TestCommandBuffer.java
@@ -1,5 +1,6 @@
 package dev.jbang.util;
 
+import static dev.jbang.util.Util.JBANG_RUNTIME_SHELL;
 import static org.hamcrest.MatcherAssert.assertThat;
 import static org.hamcrest.Matchers.*;
 
@@ -12,9 +13,25 @@ public class TestCommandBuffer extends BaseTest {
 	@Test
 	void testRunWinBat() {
 		if (Util.getOS() == Util.OS.windows) {
-			String out = Util.runCommand(examplesTestFolder.resolve("echo.bat").toString(), "abc def", "abc;def");
+			String out = Util.runCommand(examplesTestFolder.resolve("echo.bat").toString(), "abc def", "abc;def",
+					"abc=def", "abc,def");
 			assertThat(out, containsString("ARG = abc def"));
 			assertThat(out, containsString("ARG = abc;def"));
+			assertThat(out, containsString("ARG = abc=def"));
+			assertThat(out, containsString("ARG = abc,def"));
+		}
+	}
+
+	@Test
+	void testRunWinBatFromBash() {
+		if (Util.getOS() == Util.OS.windows) {
+			environmentVariables.set(JBANG_RUNTIME_SHELL, "bash");
+			String out = Util.runCommand(examplesTestFolder.resolve("echo.bat").toString(), "abc def", "abc;def",
+					"abc=def", "abc,def");
+			assertThat(out, containsString("ARG = abc def"));
+			assertThat(out, containsString("ARG = abc;def"));
+			assertThat(out, containsString("ARG = abc=def"));
+			assertThat(out, containsString("ARG = abc,def"));
 		}
 	}
 

